gsa/src/web/graphql/cves.js

```diff
@@ -0,0 +1,308 @@
+      pageInfo {
+        hasNextPage
+        hasPreviousPage
+        startCursor
+        endCursor
+        lastPageCursor
+      }
+    }
+  }
+`;
+
+export const EXPORT_CVES_BY_IDS = gql`
+  mutation exportCvesByIds($ids: [String]!) {
+    exportCvesByIds(ids: $ids) {
+      exportedEntities
+    }
+  }
+`;
+
+export const EXPORT_CVES_BY_FILTER = gql`
+  mutation exportCvesByFilter($filterString: String) {
+    exportCvesByFilter(filterString: $filterString) {
+      exportedEntities
+    }
+  }
+`;
+
+export const useExportCvesByFilter = options => {
+  const [queryExportCvesByFilter] = useMutation(EXPORT_CVES_BY_FILTER, options);
+  const exportCvesByFilter = useCallback(
+    // eslint-disable-next-line no-shadow
+    filterString =>
+      queryExportCvesByFilter({
+        ...options,
+        variables: {
+          filterString,
+        },
+      }),
+    [queryExportCvesByFilter, options],
+  );
+
+  return exportCvesByFilter;
+};
+
+export const useLazyGetCve = (id, options) => {
+  const [queryCve, {data, ...other}] = useLazyQuery(GET_CVE, {
+    ...options,
+    variables: {id},
+  });
+  const cve = isDefined(data?.cve) ? Cve.fromObject(data.cve) : undefined;
+
+  const loadCve = useCallback(
+    // eslint-disable-next-line no-shadow
+    (id, options) => queryCve({...options, variables: {id}}),
+    [queryCve],
+  );
+  return [loadCve, {cve, ...other}];
+};
+
+export const useLazyGetCves = (variables, options) => {
+  const [queryCves, {data, ...other}] = useLazyQuery(GET_CVES, {
+    ...options,
+    variables,
+  });
+  const cves = isDefined(data?.cves)
+    ? data.cves.edges.map(entity => Cve.fromObject(entity.node))
+    : undefined;
+
+  const {total, filtered, offset = -1, limit, length} =
+    data?.cves?.counts || {};
+  const counts = isDefined(data?.cves?.counts)
+    ? new CollectionCounts({
+        all: total,
+        filtered: filtered,
+        first: offset + 1,
+        length: length,
+        rows: limit,
+      })
+    : undefined;
+  const getCves = useCallback(
+    // eslint-disable-next-line no-shadow
+    (variables, options) => queryCves({...options, variables}),
+    [queryCves],
+  );
+  const pageInfo = data?.cves?.pageInfo;
+  return [getCves, {...other, counts, cves, pageInfo}];
+};
+
+export const useGetCve = (id, options) => {
+  const {data, ...other} = useQuery(GET_CVE, {
+    ...options,
+    variables: {id},
+  });
+  const cve = isDefined(data?.cve) ? Cve.fromObject(data.cve) : undefined;
+  return {cve, ...other};
+};
+
+export const useExportCvesByIds = options => {
+  const [queryExportCvesByIds] = useMutation(EXPORT_CVES_BY_IDS, options);
+
+  const exportCvesByIds = useCallback(
+    // eslint-disable-next-line no-shadow
+    cveIds =>
+      queryExportCvesByIds({
+        ...options,
+        variables: {
+          ids: cveIds,
+        },
+      }),
+    [queryExportCvesByIds, options],
+  );
+
+  return exportCvesByIds;
+};
+
+export const useLoadCvePromise = () => {
```

Comment:
We don't need this for the cves. This kind of query is used for scanconfigs because we need a lazy query that returns a promise. We need that promise so we can wait for it and work with the results before we pass them to the dialog. But that should not be necessary for cves.